src/manager.c

```
@@ -132,18 +134,22 @@ static void closep(int *fdp) {
 
 static int manager_handle_device(Manager *manager, struct udev_device *device) {
         _cleanup_(closep) int devicefd = -1, firmwarefd = -1;
+        const char *name;
         int r;
 
         devicefd = openat(manager->devicesfd, udev_device_get_syspath(device), O_RDONLY|O_NONBLOCK|O_DIRECTORY|O_CLOEXEC|O_PATH);
         if (devicefd < 0)
                 return errno == ENOENT ? 0 : -errno;
 
-        firmwarefd = manager_find_firmware(manager, udev_device_get_property_value(device, "FIRMWARE"));
+        name = udev_device_get_property_value(device, "FIRMWARE");
```

Maybe error out early if FIRMWARE is not set? This would of course be a serious kernel bug, but still... (this was already like that, so just for future reference).

---

Agreed, we need to add some more paranoia code. Also stuff like handling EINTR on close() wouldn't be a bad idea

---

Hm, handling it how? My understanding was that the best thing to do is to ignore errors on close()? https://github.com/systemd/systemd/blob/master/src/basic/fd-util.c#L37

---

Ohh, so I just learned something new. In this case no EINTR busy loop on close :) Thanks for pointing out!